tests/unit_tests/test_special_characters.py

```
@@ -0,0 +1,260 @@
+
+    def test_metadata_special_characters(self):
+        """Test that metadata keys and values with special characters are correctly handled."""
+        for special_str in self.special_chars:
+            self.client._executor.reset_mock()
+
+            # Test as value
+            metadata = {"key": special_str}
+
+            # Test as key (keys in JSON usually string, but worth testing escaping)
+            # Note: JSON keys must be strings.
+            metadata_key_test = {special_str: "value"}
```

Ruff (0.14.14)

[error] Local variable `metadata_key_test` is assigned to but never used

Remove assignment to unused variable `metadata_key_test`

(F841)